File: c/grains/src/grains.c
#include <stdio.h>

#include "grains.h"

unsigned long long square(int nr) {
    if (nr < 1 || nr > 64) {
        return 0;
    }
    return 1ull << (nr - 1);
}

unsigned long long total() {
    return ~0ull;
}

File: go/grains/grains.go
package grains

import "fmt"

// Square calculate the square of input number
func Square(input int) (uint64, error) {
	if input < 1 || input > 64 {
		return 0, fmt.Errorf("bad suqre: %d", input)
	}
	return 1 << uint(input-1), nil
}

// Total function calculate the number of grain of wheat on chessboard
func Total() (total uint64) {
	return 1<<64 - 1
}
